litmus-portal/backend/graphql-server/pkg/graphql/mutations/mutation.go

```diff
@@ -88,3 +88,50 @@ func NewEvent(clusterEvent model.ClusterEventInput, r store.StateData) (string,
 
 	return "", errors.New("ERROR WITH CLUSTER EVENT")
 }
+
+// UpsterWorkFlowRun Updates or Inserts a new Workflow Run into the DB
+func UpsterWorkFlowRun(input model.WorkflowRunInput, r store.StateData) (string, error) {
+	cluster, err := cluster.VerifyCluster(*input.ClusterID)
+	if err != nil {
+		log.Print("ERROR", err)
+		return "", err
+	}
+	newWorkflowRun := model.WorkflowRun{
+		ClusterID:     cluster.ClusterID,
+		ClusterName:   cluster.ClusterName,
+		ProjectID:     cluster.ProjectID,
+		LastUpdated:   strconv.FormatInt(time.Now().Unix(), 10),
+		WorkflowRunID: input.WorkflowRunID,
+		WorkflowName:  input.WorkflowName,
+		ExecutionData: input.ExecutionData,
+		WorkflowID:    "000000000000",
+	}
+	subscriptions.SendWorkflowEvent(newWorkflowRun, r)
+	err = database.UpsertWorkflowRun(newWorkflowRun)
+	if err != nil {
+		log.Print("ERROR", err)
+		return "", err
+	}
+	return "Workflow Run Accepted", nil
+}
+
+// LogsHandler receives logs from the workflow-agent and publishes to frontend clients
+func LogsHandler(podLog model.PodLog, r store.StateData) (string, error) {
+	_, err := cluster.VerifyCluster(*podLog.ClusterID)
+	if err != nil {
+		log.Print("ERROR", err)
+		return "", err
+	}
+	if reqChan, ok := r.WorkflowLog[podLog.RequestID]; ok {
+		resp := model.PodLogResponse{
+			PodName:       podLog.PodName,
+			WorkflowRunID: podLog.WorkflowRunID,
+			PodType:       podLog.PodType,
+			Log:           podLog.Log,
+		}
+		reqChan <- &resp
+		close(reqChan)
+		return "LOGS SENT SUCCESSFULLY", nil
+	}
+	return "LOG REQUEST CANCELLED", nil
```

Comment:
This return will throw an error, but it is nil. Make to one parameter response

Reply:
We need both a string return and a error return because the output is used by the GQL resolver which requires a string(response) and error

Reply:
You have to throw an error here

Reply:
No it is not an error, so basically the frontend can cancel the log request anytime they want as it is on-demand and throwing an error here isn't required as the agent doesn't need to know about this.